query/src/servers/http/v1/multipart_format.rs

```diff
@@ -246,6 +269,47 @@ impl Processor for SequentialInputFormatSource {
         let mut progress_values = ProgressValues::default();
         match replace(&mut self.state, State::NeedReceiveData) {
             State::ReceivedData(data) => {
+                let data = match &mut self.input_decompress {
+                    None => data,
+                    Some(decompress) => {
+                        let mut output = Vec::new();
+                        let mut amt = 0;
+
+                        loop {
```

Comment:
Why we need this?

Reply:
DecompressDecode itself is a state machine, and we don't know how much data it will output. So will loop here to make sure the whole decompress process is done, or we already decoded all data and need to fetch more.